### PR TITLE
allow redirecting metarange and range requests to presigned URLs

### LINKED ISSUE #7781: improvement: add pre-signed url support to metarange/range get endpoints
allow redirecting to a presigned URL if `?presign=true`.

### PR DESCRIPTION
**Pull request by ozkatz**
closes #7781

### CODE REVIEW

**Comment by ozkatz** [comment]
closes #7781 7781

File: pkg/api/controller.go [hunk 1]
```diff
@@ -3684,14 +3684,34 @@ func (c *Controller) GetMetaRange(w http.ResponseWriter, r *http.Request, reposi
 		return
 	}
 
+	// check if presign
+	if swag.BoolValue(params.Presign) {
+		repo, err := c.Catalog.GetRepository(ctx, repository)
+		if c.handleAPIError(ctx, w, r, err) {
+			return
+		}
```
Comment (guy-har) [inline pkg/api/controller.go:3692]: Can we use the response returned from `c.Catalog.GetMetarange` ?
If not please move this before the call to `GetMetarange`

Comment (ozkatz) [reply pkg/api/controller.go:3692]: The existing (non-presigned) flow doesn't require fetching the repository object. Moving it means adding an additional lookup in other flows. Is there a specific reason to do so?

**Comment by ozkatz** [comment]
After discussing with @guy-har - closing this PR for now. Will open a separate internal endpoint to download meta-ranges and ranges (with/without presign)

**Pull request closed without merging**